bin/wasm-node/javascript/src/worker/bindings-smoldot-light.ts

```diff
@@ -0,0 +1,310 @@
+// Smoldot
+// Copyright (C) 2019-2022  Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Exports a function that provides bindings for the bindings found in the Rust part of the code.
+//!
+//! In order to use this code, call the function passing an object, then fill the `instance` field
+//! of that object with the Wasm instance.
+
+import { Buffer } from 'buffer';
+import Websocket from 'websocket';
+import * as compat from '../compat/index.js';
+import type { SmoldotWasmInstance } from './bindings.js';
+
+import type { Socket as TcpSocket } from 'net';
+
+export interface Config {
+    instance?: SmoldotWasmInstance,
+    logCallback: (level: number, target: string, message: string) => void,
+    jsonRpcCallback: (response: string, chainId: number) => void,
+    databaseContentCallback: (data: string, chainId: number) => void,
+    forbidTcp: boolean,
+    forbidWs: boolean,
+    forbidNonLocalWs: boolean,
+    forbidWss: boolean,
+}
+
+export default function (config: Config): compat.WasmModuleImports {
+    // Used below to store the list of all connections.
+    // The indices within this array are chosen by the Rust code.
+    let connections: Record<number, TcpWrapped | WebSocketWrapped> = {};
+
+    return {
+        // Must exit with an error. A human-readable message can be found in the WebAssembly
+        // memory in the given buffer.
+        panic: (ptr: number, len: number) => {
+            const instance = config.instance as SmoldotWasmInstance;
```

Comment:
Perhaps a better way of doing this would be:

```ts
const intance = config.instance!
```

Reply:
or just
```ts
const { instance } = config
```
if the Config interface defines the instance as a non-optional field.